include/solvers/time_solver.h

```diff
@@ -261,12 +280,33 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
   { _is_adjoint = _is_adjoint_value; }
 
   /**
-   * Function to return 'last_deltat()', returns system.deltat if
-   * fixed timestep solver is used, completed_deltat if the adaptive
+   * Returns system.deltat if fixed timestep solver is used,
+   * the complete timestep size (sum of all substeps) if the adaptive
```

Comment:
This is just confusing to anyone who hasn't been buried in TwoStepTimeSolver for a month.  As far as we're concerned at this level of abstraction, the time step length is deltat.  Maybe there's some high p trickery going on, or it's been subdivided under the hood, or we tried some different deltats and settled on one that worked, but that's supposed to be effectively hidden at the level of the base class.  "Returns the change in system.time, deltat, for the last timestep which was successfully completed."  You could then elaborate with "This only returns the outermost step size in the case of nested time solvers." if you want.  But you should also add "If no time step has yet been successfully completed, then returns X", because that's not yet defined by any version of this comment, and at this point you're the only person who knows what X is.  :-D

Reply:
We initialize last_deltat and completed_timestep_size with s.deltat  in the constructor now.